src/core/server/http/http_service.ts

```diff
@@ -20,13 +20,17 @@
 import { Observable, Subscription } from 'rxjs';
 import { first } from 'rxjs/operators';
 
-import { CoreService } from '../../types/core_service';
+import { CoreService } from '../../types';
 import { Logger, LoggerFactory } from '../logging';
 import { HttpConfig } from './http_config';
 import { HttpServer, HttpServerInfo } from './http_server';
 import { HttpsRedirectServer } from './https_redirect_server';
 import { Router } from './router';
 
+/** @internal */
+export type HttpServiceStartContract = HttpServerInfo;
```

Comment:
This isn't really related to this PR, but I wanted to raise the concern while I saw it so we can address it later: I don't like that `HttpServiceStartContract` exposes hapi's server object. It seems like we're doing it with `HttpServerInfo` just so we can support the legacy integration with KbnServer, and that's OK with me because `HttpServerInfo` is an internal detail to the core system. But we should treat the contracts we return from the service lifecycle functions as public interfaces and ultimately expose them as such to the corresponding lifecycle function on plugins. This convention gives us consistency across services for what functionality gets exposed to plugins, and within core itself you can always pass the entire service to another service if you want to take advantage of core-specific details.

Reply:
Yeah, totally agree, there is no doubt that we shouldn't ever expose something like this (Joi, Hapi or any other internals). We make this nasty exception to support compatibility bridge between new and legacy HTTP servers.